activesupport/test/core_ext/object/to_query_test.rb

```diff
@@ -91,6 +96,30 @@ def test_hash_not_sorted_lexicographically_for_nested_structure
     assert_equal expected, URI.decode_www_form_component(params.to_query)
   end
 
+  def test_hash_not_sorted_lexicographically_if_option_specified
+    hash = { type: "human", name: "Nakshay" }
+    assert_equal "type=human&name=Nakshay", hash.to_query(preserve_order: true)
+  end
+
+  def test_hash_sorted_lexicographically_if_option_specified
+    hash = { type: "human", name: "Nakshay" }
+    assert_equal "name=Nakshay&type=human", hash.to_query(preserve_order: false)
+  end
+
+  def test_hash_with_namespace_and_preserve_order_options_provided
+    hash = { type: "human", name: "Nakshay" }
+    expected = "earth[type]=human&earth[name]=Nakshay"
+    actual = hash.to_query(preserve_order: true, namespace: "earth")
+    assert_equal expected, URI.decode_www_form_component(actual)
+  end
+
+  def test_hash_as_a_key
```

Comment:
This is a test that would not pass in the related pull request I reference in this pull request's description.